sklearn/utils/validation.py

```diff
@@ -1292,6 +1376,14 @@ def check_X_y(
             f"{estimator_name} requires y to be passed, but the target y is None"
         )
 
+    if force_all_finite != "deprecated":
```

Comment:
What should the behavior be when both are set?

```python
check_array(force_all_finite=True, ensure_all_finite=False)
```

Reply:
The way I implemented it for now is that if both are set, `force_all_finite` takes over.
I can see that we'd like to have an error if both are different though. Let me change that

Reply:
Instead, I added a comment in warning saying that `ensure_all_finite` is ignored when `force_all_finite`. I know that since `ensure_all_finite` is the new name it feels a bit weird and one would expect the opposite, but this way keeps things simple and avoids adding a complex logic.

What do you think ?

Reply:
I'm +0.5 on just improving the docs/warning message.

The alternative is to have `ensure_all_finite=None` as the default that behaves as `True`. Then if `force_all_finite!="deprecated"` and  `ensure_all_finite is not None`, raise an error that says that are not be set at the same time?

In two versions, we update `ensure_all_finite=True` and it'll be backward compatible because `None` behaved the same way.

Reply:
> The alternative is to have `ensure_all_finite=None`

Yes that's what I was not enthusiastic about. I would be in favor of doing that only if we agree to remove `None` as a valid option in 1.8 without deprecation :smile:. Because otherwise I find that a 4 releases cycle is too long for a renaming.

Reply:
Since this is a developer tools, I think I'm fine avoiding to have the default sentinel `None` and later change the default.

Reply:
I'm not sure what @glemaitre is suggesting here. But I would rather use `None`, not document it, and later remove it after deprecation of `force_...`

Reply:
I changed it so that now an error is raised when both are set. I added a comment to remove `None` in 1.8 without deprecation.